src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -191,8 +252,14 @@ private ValueTask<VoidResult> ProcessEventChangeAsync(ImmutableSegmentedList<boo
         /// <param name="highPriority">
         /// If this tagging request should be processed as quickly as possible with no extra delays added for it.
         /// </param>
-        private async Task<VoidResult> RecomputeTagsAsync(bool highPriority, CancellationToken cancellationToken)
+        private async Task<VoidResult> RecomputeTagsAsync(
+            bool highPriority,
+            bool frozenPartialSemantics,
+            CancellationToken cancellationToken)
         {
+            if (cancellationToken.IsCancellationRequested)
+                return default;
```

Comment:
The `SwitchToMainThreadAsync` operation on the next line will complete immediately for this case, so this is effectively an unnecessary duplicate of the check two lines down.